// @generated/type-graphql/resolvers/relations/Schema/SchemaRelationsResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaRelationsResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
const Attestation_1 = require("../../../models/Attestation");
const Schema_1 = require("../../../models/Schema");
const SchemaName_1 = require("../../../models/SchemaName");
const SchemaAttestationsArgs_1 = require("./args/SchemaAttestationsArgs");
const SchemaSchemaNamesArgs_1 = require("./args/SchemaSchemaNamesArgs");
const helpers_1 = require("../../../helpers");
let SchemaRelationsResolver = class SchemaRelationsResolver {
    async attestations(schema, ctx, info, args) {
        const { _count } = (0, helpers_1.transformInfoIntoPrismaArgs)(info);
        return (0, helpers_1.getPrismaFromContext)(ctx).schema.findUniqueOrThrow({
            where: {
                id: schema.id,
            },
        }).attestations({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async schemaNames(schema, ctx, info, args) {
        const { _count } = (0, helpers_1.transformInfoIntoPrismaArgs)(info);
        return (0, helpers_1.getPrismaFromContext)(ctx).schema.findUniqueOrThrow({
            where: {
                id: schema.id,
            },
        }).schemaNames({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
};
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [Attestation_1.Attestation], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Info()),
    tslib_1.__param(3, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Schema_1.Schema, Object, Object, SchemaAttestationsArgs_1.SchemaAttestationsArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], SchemaRelationsResolver.prototype, "attestations", null);
tslib_1.__decorate([
    TypeGraphQL.FieldResolver(_type => [SchemaName_1.SchemaName], {
        nullable: false
    }),
    tslib_1.__param(0, TypeGraphQL.Root()),
    tslib_1.__param(1, TypeGraphQL.Ctx()),
    tslib_1.__param(2, TypeGraphQL.Info()),
    tslib_1.__param(3, TypeGraphQL.Args()),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", [Schema_1.Schema, Object, Object, SchemaSchemaNamesArgs_1.SchemaSchemaNamesArgs]),
    tslib_1.__metadata("design:returntype", Promise)
], SchemaRelationsResolver.prototype, "schemaNames", null);
SchemaRelationsResolver = tslib_1.__decorate([
    TypeGraphQL.Resolver(_of => Schema_1.Schema)
], SchemaRelationsResolver);
exports.SchemaRelationsResolver = SchemaRelationsResolver;
