// @generated/type-graphql/models/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Timestamp = exports.ServiceStat = exports.SchemaName = exports.Schema = exports.OffchainRevocation = exports.EnsName = exports.Attestation = void 0;
var Attestation_1 = require("./Attestation");
Object.defineProperty(exports, "Attestation", { enumerable: true, get: function () { return Attestation_1.Attestation; } });
var EnsName_1 = require("./EnsName");
Object.defineProperty(exports, "EnsName", { enumerable: true, get: function () { return EnsName_1.EnsName; } });
var OffchainRevocation_1 = require("./OffchainRevocation");
Object.defineProperty(exports, "OffchainRevocation", { enumerable: true, get: function () { return OffchainRevocation_1.OffchainRevocation; } });
var Schema_1 = require("./Schema");
Object.defineProperty(exports, "Schema", { enumerable: true, get: function () { return Schema_1.Schema; } });
var SchemaName_1 = require("./SchemaName");
Object.defineProperty(exports, "SchemaName", { enumerable: true, get: function () { return SchemaName_1.SchemaName; } });
var ServiceStat_1 = require("./ServiceStat");
Object.defineProperty(exports, "ServiceStat", { enumerable: true, get: function () { return ServiceStat_1.ServiceStat; } });
var Timestamp_1 = require("./Timestamp");
Object.defineProperty(exports, "Timestamp", { enumerable: true, get: function () { return Timestamp_1.Timestamp; } });
