// @generated/type-graphql/enums/AttestationScalarFieldEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttestationScalarFieldEnum = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = tslib_1.__importStar(require("type-graphql"));
var AttestationScalarFieldEnum;
(function (AttestationScalarFieldEnum) {
    AttestationScalarFieldEnum["id"] = "id";
    AttestationScalarFieldEnum["data"] = "data";
    AttestationScalarFieldEnum["decodedDataJson"] = "decodedDataJson";
    AttestationScalarFieldEnum["recipient"] = "recipient";
    AttestationScalarFieldEnum["attester"] = "attester";
    AttestationScalarFieldEnum["time"] = "time";
    AttestationScalarFieldEnum["timeCreated"] = "timeCreated";
    AttestationScalarFieldEnum["expirationTime"] = "expirationTime";
    AttestationScalarFieldEnum["revocationTime"] = "revocationTime";
    AttestationScalarFieldEnum["refUID"] = "refUID";
    AttestationScalarFieldEnum["revocable"] = "revocable";
    AttestationScalarFieldEnum["revoked"] = "revoked";
    AttestationScalarFieldEnum["txid"] = "txid";
    AttestationScalarFieldEnum["schemaId"] = "schemaId";
    AttestationScalarFieldEnum["ipfsHash"] = "ipfsHash";
    AttestationScalarFieldEnum["isOffchain"] = "isOffchain";
})(AttestationScalarFieldEnum = exports.AttestationScalarFieldEnum || (exports.AttestationScalarFieldEnum = {}));
TypeGraphQL.registerEnumType(AttestationScalarFieldEnum, {
    name: "AttestationScalarFieldEnum",
    description: undefined,
});
